src/main/kotlin/no/nav/syfo/domain/Sykmelding.kt

```diff
@@ -64,6 +70,11 @@ enum class Periodetype {
     REISETILSKUDD,
 }
 
+data class Diagnose(
```

Comment:
Vi har `Diagnose(internmodell) | Brukerdiagnose(utgående) | DiagnoseDTO(innkommende)` det er litt tunga rett i munnen. Jeg tror jeg blir mest forvirret av bruker*prefixet. Vi har brukt RS tidligere for å tydeliggjøre at det er transportobjekter for restlaget. Kunne det vært en vurdering å innføre det her også?

Reply:
Hva er funksjonen til `Brukerssykmelding` modellen? Er det en grunn til at queriene ikke kan returnere en `Sykmelding` direkte?

Reply:
DTO er rest-lag, utgående. Bruker er det vi henter fra databasen. Men det er kanskje feil? Jeg er litt usikker på hvordan det burde organiseres og har egentlig tenkt at vi burde ta en liten snakk om det. Jeg kaller det brukersykmelding siden det er en brukers innsyn i en sykmelding. Det interne objektet gir ikke noe serlig verdi annet enn å være et lag mellom db og rest.

Reply:
Ja, jeg tror ikke vi trenger en egen datamodell for databaselaget. Der holder det nok med mappere fra `Sykmelding -> spørring`. Da får du typisk signaturer som:

Sykmeldingservice.hentSykmeldinger(aktor:String) [her kommer typisk tilgangskontroll osv]
sykmeldingQueries.hentSykmeldinger(aktor:String) -> [Sykmelding]

og

Sykmeldingservice.bekreftSykmelding(id:String) [tilgangskontroll, sanitycheck osv] ->
sykmeldingQueries.bekreftSykmelding(id:String) -> void | Sykmelding

Poenget er at f.eks endringen av datatypen til et felt i en databasetabell skal skal være helt transparent for servicelogikken, og da automatisk for restlaget (siden det er lagdelt). Og tilsvarende skal en endring i restlaget være helt likegyldig for databasen. 

Vi kan se på det i morgen!